### PR TITLE
examples(coffeescript): update module

### LINKED ISSUE #5423: Coffeescript Vuex silently stop working on 2.5.1
### Version

[v2.5.1](https://github.com/nuxt.js/releases/tag/v2.5.1)

### Reproduction link

[https://github.com/nuxt/nuxt.js/tree/dev/examples/coffeescript](https://github.com/nuxt/nuxt.js/tree/dev/examples/coffeescript)

### Steps to reproduce

Build project with `"nuxt":  "2.5.1"`

### What is expected ?

Vuex that works with Coffeescript, just like on 2.4.5

### What is actually happening?

Nothing.

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8960">#c8960</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by devartyom**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Resolves [#5423](https://github.com/nuxt/nuxt.js/issues/5423)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

File: packages/builder/src/builder.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ export default class Builder {
       restart: null
     }
 
-    this.supportedExtensions = ['vue', 'js', ...(this.options.build.additionalExtensions || [])]
+    this.supportedExtensions = ['vue', 'js', ...(this.options.extensions || []), ...(this.options.build.additionalExtensions || [])]
```
Comment (pi0) [inline packages/builder/src/builder.js]: Are these options identical now? /cc @kevinmarrec

Comment (DevArtyom) [reply packages/builder/src/builder.js]: Please let me know if I should use nuxt.options.build.additionalExtensions in the module that add support for coffeescript extension.
Like this.nuxt.options.build.additionalExtensions.push ('coffee')

Comment (kevinmarrec) [reply packages/builder/src/builder.js]: @pi0 No
https://github.com/nuxt/typescript/blob/master/packages/typescript-build/lib/module.js#L18
We need to be aware that the 2 options aren't used at the same places in core

Comment (kevinmarrec) [reply packages/builder/src/builder.js]: `options.extensions` is more or less related to runtime, meanwhile the others only buildtime

**Comment by clarkdo** [comment]
Any reason for not using additionalExtensions?

**Comment by kevinmarrec** [comment]
@pi0 @clarkdo If I created a new option it's cause there was a particular need.

All i know is that `options.extensions` is used to resolves files : 
https://github.com/nuxt/nuxt.js/blob/dev/packages/config/src/options.js#L164
https://github.com/nuxt/nuxt.js/blob/dev/packages/core/src/resolver.js#L95

And `options.build.additionalExtensions` is used to build files that are not `.vue` or `.js` around Builder, or mostly create the router from files having these extensions.

This requested change https://github.com/nuxt/nuxt.js/pull/6248/files#diff-4f235fc89b94513b32c2cc8b5896cc5aR47 , would make `js` duplicated + make Builder build `.mjs` which Idk if it's relevant.

Having only one `extensions` would mean that the TypeScript module would make Nuxt also trying to resolve `.tsx` which shouldn't be needed ?

Idk

**Comment by pi0** [comment]
@kevinmarrec Seems reasonable to have separated ones because of `tsx` :)

**Review by pi0 — DISMISSED** [review]

**Comment by clarkdo** [comment]
Sorry, I didn’t mean that we need to keep only one config, I mean that why we need to merge extensions into supportedExtensions instead of configuring additionalExtensions. For the coffeescript case, only adding this won’t work, user still needs to setup loader.  And current change also changed the resolving order and files of routes, it’s kind of breaking change for users having files inside pages dir but not meant to be a route.

**Comment by kevinmarrec** [comment]
Exactly @clarkdo, cofeescript support for routes should laverage `build.additionalExtensions` within the cofeescript module.

If somehow cofeescript files should be resolved for other files like serverMiddlewares or local modules, then it should be done like TypeScript module by using both of config extensions options :
https://github.com/nuxt/typescript/blob/master/packages/typescript-build/lib/module.js#L18

**Comment by pi0** [comment]
The breaking change seems introduced by 2.4.5...2.51 -- we are reverting :)

**Comment by kevinmarrec** [comment]
@pi0 AFAIK routes supported extensions never used options. It was hardcoded, what breaking change was it ?

**Comment by pi0** [comment]
According to #5423 CoffeeScript support was working and stopped working. But yeah you are right it was hardcoded. [srec](https://github.com/nuxt/nuxt.js/blob/v2.4.0/packages/builder/src/builder.js#L50] so strange...

**Comment by kevinmarrec** [comment]
According the issue, a partial fix was found making Vuex World but not local modules, which kind of relevant as to have coffee work everywhere it nows need to be added in both extensions options

I think the fix it just about adding `.coffee` extension to `build.additionalExtensions` and nothing more.

There was anyway Indeed something changed in internals. Once upon a time there was a `resolveFiles` that seemed merging `options.extensions` inside `supportedExtensions` of the class that was storint this `resolveFiles` function

**Comment by kevinmarrec** [comment]
@pi0 LGTM now, juste need title update

**Review by kevinmarrec — APPROVED** [review]

**Review by clarkdo — APPROVED** [review]

**Pull request merged**